### PR TITLE
Fix bug: The '?:' operator has a lower priority than the '==' operator.

### LINKED ISSUE #62722: Looks very suspicious. The '?:' operator has a lower priority than the '==' operator.  Bug?
https://github.com/apple/swift/blob/4cf6f79938c1fecd0ea458110ce54d16d0470a00/lib/Frontend/ArgsToFrontendOutputsConverter.cpp#L284

This assert looks very suspicious.
```C++
assert(OutputFiles.size() == InputsAndOutputs.inputCount() &&
            pathsFromUser->size() == InputsAndOutputs.hasInputs()
               ? 1
               : 0);
```

Shouldn't this be following?
```C++
assert(OutputFiles.size() == InputsAndOutputs.inputCount() &&
            pathsFromUser->size() == (InputsAndOutputs.hasInputs() ? 1 : 0));
```

### PR DESCRIPTION
**Pull request by valeriyvan**
Resolves #62722.

### CODE REVIEW

**Comment by valeriyvan** [comment]
@CodaFi, may I ask for review?

**Comment by valeriyvan** [comment]
ping

**Comment by valeriyvan** [comment]
ping

**Comment by valeriyvan** [comment]
ping

File: lib/Frontend/ArgsToFrontendOutputsConverter.cpp [hunk 1]
```diff
@@ -282,9 +282,7 @@ SupplementaryOutputPathsComputer::computeOutputPaths() const {
   else {
     // Multi-threaded WMO is the exception
     assert(OutputFiles.size() == InputsAndOutputs.inputCount() &&
-                   pathsFromUser->size() == InputsAndOutputs.hasInputs()
-               ? 1
-               : 0);
+                   pathsFromUser->size() == (InputsAndOutputs.hasInputs() ? 1 : 0));
```
Comment (AnthonyLatsis) [inline lib/Frontend/ArgsToFrontendOutputsConverter.cpp:285]: Could you `clang-format` this?

**Comment by AnthonyLatsis** [comment]
@CodaFi ping

File: lib/Frontend/ArgsToFrontendOutputsConverter.cpp (hunk 1 above)
Comment (valeriyvan) [reply lib/Frontend/ArgsToFrontendOutputsConverter.cpp:285]: output from `clang-format:
```
           pathsFromUser->size() == (InputsAndOutputs.hasInputs() ? 1 : 0));
```

Comment (valeriyvan) [reply lib/Frontend/ArgsToFrontendOutputsConverter.cpp]: > Yes, so the column of both && operands should match.

No, they shouldn't as strings before && don't match.
I've changed indentation to format assert exactly like one above it.

Comment (AnthonyLatsis) [reply lib/Frontend/ArgsToFrontendOutputsConverter.cpp]: > I've changed indentation to format assert exactly like one above it.

The indentation is now correct, thanks. I think you misunderstood my point though: I was reasserting, not contradicting the formatter output, that both `&&` operands (`OutputFiles.size() == InputsAndOutputs.inputCount()` and `pathsFromUser->size() == (InputsAndOutputs.hasInputs() ? 1 : 0)`) should be aligned by column. (as they are now)

**Review by akyrtzi — APPROVED** [review]

**Comment by akyrtzi** [comment]
@swift-ci smoke test and merge

**Comment by AnthonyLatsis** [comment]
@akyrtzi Thank you!

**Pull request merged**